src/openrct2-ui/WindowManager.cpp

```diff
@@ -184,9 +184,9 @@ class WindowManager final : public IWindowManager
             case WD_BANNER:
                 return window_banner_open(id);
             case WD_DEMOLISH_RIDE:
-                return window_ride_demolish_prompt_open(get_ride(id));
+                return window_ride_demolish_prompt_open(get_ride(static_cast<ride_id_t>(id)));
```

Comment:
I feel we should just bubble up the change into the callers of OpenDetails

Reply:
If we provide strong types for the others we could simply add overloads for each and also use variants on the window but I think this sort of clean up would be the last step after having the types.